tests/nested/manual/fakeinstaller/task.yaml

```diff
@@ -0,0 +1,165 @@
+  #shellcheck source=tests/lib/nested.sh
+  . "$TESTSLIB"/nested.sh
+  version="$(nested_get_version)"
+  # get an updated kernel
+  snap download --basename=pc-kernel --channel="$version/edge" pc-kernel
+  uc20_build_initramfs_kernel_snap "$PWD/pc-kernel.snap" "$NESTED_ASSETS_DIR"
+  mv "${NESTED_ASSETS_DIR}"/pc-kernel_*.snap pc-kernel.snap
+
+  # prepare a classic seed
+  # TODO:
+  # - create pc-classic custom gadget
+  # - repacked snapd snap
+  # (should be as simple as addinga "--snap=./local-gadget.snap ...")
+  LABEL="$(date +%Y%m%d)"
+  gendeveloper1 sign-model < "$TESTSLIB"/assertions/developer1-22-classic-dangerous.json > my.model
+  snap prepare-image --classic \
+      --channel=edge \
+      --snap ./pc-kernel.snap \
+      my.model \
+      ./classic-seed
+  cp -a ./classic-seed/system-seed/ /var/lib/snapd/seed
+
+  # do some light checking that the system is valid
+  test-snapd-curl.curl -s --unix-socket /run/snapd.socket http://localhost/v2/systems | jq '.result.systems[0].label' | MATCH "$LABEL"
+  test-snapd-curl.curl -s --unix-socket /run/snapd.socket http://localhost/v2/systems/"$LABEL" > system
+  jq '.result.model.distribution' system | MATCH "ubuntu"
+
+  # build fakeinstaller and put in place
+  go build -o fakeinstaller "$TESTSLIB"/fakeinstaller/main.go
+
+  # create fake disk for the installer to work on
+  truncate --size=4G fake-disk.img
+  loop_device=$(losetup --show -f ./fake-disk.img)
+  # TODO: fakeinstaller should work on a empty disk too
+  echo "label: gpt" | sfdisk "$loop_device"
+  # and "install" the current seed to the fake disk
+  ./fakeinstaller "$LABEL" "$loop_device" "$TESTSLIB"/fakeinstaller/mk-classic-rootfs.sh
+  # validate that the fake installer created the expected partitions
+  fdisk -x "$loop_device" > fdisk_output
+  MATCH "${loop_device}p1 .* BIOS Boot"   < fdisk_output
+  # TODO: the real MVP hybrid device will not contain a ubuntu-seed
+  #       partition (needs a different gadget)
+  MATCH "${loop_device}p2 .* ubuntu-seed" < fdisk_output
+  MATCH "${loop_device}p3 .* ubuntu-boot" < fdisk_output
+  MATCH "${loop_device}p4 .* ubuntu-save" < fdisk_output
+  MATCH "${loop_device}p5 .* ubuntu-data" < fdisk_output
+
+  # image partitions are not mounted anymore
+  for d in ubuntu-seed ubuntu-boot ubuntu-data ubuntu-save; do
+      test -d /run/mnt/"$d"
+      not mountpoint /run/mnt/"$d"
+  done
+
+  # mount image to inspect data
+  mount -o ro "${loop_device}"p3 /run/mnt/ubuntu-boot
+  mount -o ro "${loop_device}"p5 /run/mnt/ubuntu-data
+
+  # seed is populated
+  test -d /run/mnt/ubuntu-data/var/lib/snapd/seed/systems/"$LABEL"
+  # rootfs is there
+  test -x /run/mnt/ubuntu-data/usr/lib/systemd/systemd
+  # ensure not "ubuntu-data/system-data" is generated, this is a dir only
+  # used on core and should not be there on classic
+  #not test -d /run/mnt/ubuntu-data/system-data
+  # TODO: ensure we don't have this
+  #not test -d /run/mnt/ubuntu-data/_writable_defaults
```

Comment:
This can be enabled now :) AFAICT your code DTRT and the dir is not there on classic.